### PR TITLE
feat: add live logs list

### PR DESCRIPTION
**Pull request by yeshev**
part of #3133

### CODE REVIEW

**Review by nityanandagohain — COMMENTED** [review]
* When no log lines are fetched, it should show something like  "fetching log lines" instead of "no log lines found" cc @palashgdev

**Comment by yeshev** [comment]
> 

@nityanandagohain Fixed

**Review by nityanandagohain — CHANGES_REQUESTED** [review]
@yeshev 
* When I resume the id sent is not correct it seems, it should send the id of the last log line received, i.e log line at the top. but it is sending something else.
* The same issue is with search, the id send in the API request is not correct.

**Review by nityanandagohain — APPROVED** [review]

File: frontend/src/container/LiveLogs/ListViewPanel/index.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import { DataSource, StringOperators } from 'types/common/queryBuilder';
+
+function ListViewPanel(): JSX.Element {
+	const { config } = useOptionsMenu({
+		storageKey: LOCALSTORAGE.LOGS_LIST_OPTIONS,
+		dataSource: DataSource.LOGS,
+		aggregateOperator: StringOperators.NOOP,
+	});
+
+	const isFormatButtonVisible = useMemo(
```
Comment (YounixM) [inline frontend/src/container/LiveLogs/ListViewPanel/index.tsx:20]: useMemo isn't required here.

File: frontend/src/container/LiveLogs/LiveLogsContainer/index.tsx [hunk 2]
```diff
@@ -117,42 +120,49 @@ function LiveLogsContainer(): JSX.Element {
...
 	useEffect(() => {
-		if (compositeQuery?.id !== stagedQuery?.id || initialLoading) {
+		if (!compositeQuery || !stagedQuery) return;
+
+		if (compositeQuery.id !== stagedQuery.id || initialLoading) {
 			handleStartNewConnection();
 		}
 	}, [stagedQuery, initialLoading, compositeQuery, handleStartNewConnection]);
 
+	console.log({ logs });
```
Comment (YounixM) [inline frontend/src/container/LiveLogs/LiveLogsContainer/index.tsx:137]: Remove console log

File: frontend/src/container/LiveLogs/LiveLogsContainer/index.tsx [hunk 3]
```diff
@@ -117,42 +120,49 @@ function LiveLogsContainer(): JSX.Element {
...
+		<Wrapper>
+			<LiveLogsTopNav />
+			<ContentWrapper gutter={[0, 20]} style={{ color: themeColors.lightWhite }}>
 				<Col span={24}>
 					<FiltersInput />
 				</Col>
 				{initialLoading ? (
 					<Col span={24}>
-						<Spinner style={{ height: 'auto' }} />
+						<Spinner style={{ height: 'auto' }} tip="Getting logs" />
```
Comment (YounixM) [inline frontend/src/container/LiveLogs/LiveLogsContainer/index.tsx:148]: tip can be updated to `Fetching Logs`

File: frontend/src/container/LiveLogs/LiveLogsList/index.tsx [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+				weight: '300',
+			},
+		],
+		options.format === 'raw',
+		{
+			timeout: 5000,
+		},
+	);
+
+	const selectedFields = useMemo(
```
Comment (YounixM) [inline frontend/src/container/LiveLogs/LiveLogsList/index.tsx:54]: useMemo isn't required here.

https://blog.logrocket.com/when-not-to-use-usememo-react-hook/#:~:text=It's%20important%20to%20note%20that,itself%20has%20a%20small%20overhead.

File: frontend/src/container/LiveLogs/LiveLogsList/index.tsx [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+		() => convertKeysToColumnFields(options.selectColumns),
+		[options],
+	);
+
+	const getItemContent = useCallback(
```
Comment (YounixM) [inline frontend/src/container/LiveLogs/LiveLogsList/index.tsx:59]: Can you please profile the performance with and without using useCallback and we can then take a call which one to go with.

File: frontend/src/container/LiveLogs/LiveLogsList/index.tsx [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+		if (!activeLogId || activeLogIndex < 0) return;
+
+		ref?.current?.scrollToIndex({
+			index: activeLogIndex,
+			align: 'start',
+			behavior: 'smooth',
+		});
+	}, [activeLogId, activeLogIndex]);
+
+	const renderContent = useMemo(() => {
```
Comment (YounixM) [inline frontend/src/container/LiveLogs/LiveLogsList/index.tsx:84]: We can move this as another component and use React.memo to have good performance.

File: frontend/src/container/LiveLogs/LiveLogsList/index.tsx [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+					useWindowScroll
+					data={logs}
+					totalCount={logs.length}
+					itemContent={getItemContent}
+				/>
+			</Card>
+		);
+	}, [logs, options.format, options.maxLines, getItemContent, selectedFields]);
+
+	const isLoadingList = useMemo(() => isConnectionLoading && logs.length === 0, [
```
Comment (YounixM) [inline frontend/src/container/LiveLogs/LiveLogsList/index.tsx:113]: useMemo isn't required here.

https://blog.logrocket.com/when-not-to-use-usememo-react-hook/#:~:text=It's%20important%20to%20note%20that,itself%20has%20a%20small%20overhead.

File: frontend/src/container/LiveLogs/LiveLogsList/index.tsx [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+		isConnectionLoading,
+	]);
+
+	if (isLoadingList) {
+		return <Spinner style={{ height: 'auto' }} tip="Getting logs" />;
+	}
+
+	return (
+		<>
+			{options.format !== 'table' && (
```
Comment (YounixM) [inline frontend/src/container/LiveLogs/LiveLogsList/index.tsx:124]: We can use constants for optionsFormat from `frontend/src/constants/optionsFormatTypes.ts`

File: frontend/src/container/LiveLogs/LiveLogsList/index.tsx [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+				<Heading>
+					<Typography.Text>Event</Typography.Text>
+				</Heading>
+			)}
+
+			{logs.length === 0 && <Typography>Fetching log lines</Typography>}
```
Comment (YounixM) [inline frontend/src/container/LiveLogs/LiveLogsList/index.tsx:130]: Move `Fetching log lines` to locale file.

**Review by palashgdev — APPROVED** [review]

**Pull request merged**